### DISCUSSION TITLE
Welcome to bincraft Discussions!

### DISCUSSION THREAD

**Discussion by Escapingbug** [StarCrossPortal/sleighcraft#7] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by gamous** [comment]
Hello! I'm excited to see a awesome toolchain being developed, but I do have some ideas I'd like to discuss with you.
Is the IR mentioned by readme is PCode ?
I think it also needs more warping or transforming to make it meaningful. It's good IR inside the software implementation. However, I think its readability as a result is too poor to improve efficiency.
Do you have ideas to convert it to LLVM IR and build AST from this? (It maybe tough but necessary if we want a unique decompiler.
Or just translate the original binary code to target architecture and continue dragging to the existing tool?
I was excited when I saw QueryCraft,  but as mentioned above,  it look like just a query to pcode. In my imagination, more could have been done. If the original instructions and addresses can be structured and loaded together with the instruction meta-information in Pcode, gadgets can be queried. If AST reconstruction is done at the same time, can we do variation analysis in the gray box like CodeQL?
I didn't see the above content in the update plan, so I want to ask. ( Thonk
